From 9a9e2df7b67c970bb95df77ac63a55e86f7ab4a3 Mon Sep 17 00:00:00 2001
From: Ravi Babu <ravibabu@ti.com>
Date: Fri, 22 Mar 2013 17:51:53 +0530
Subject: [PATCH 21/30] usb: musb: host: reset dma->actual_len after transfer complete

after the dma transfer complete, reset the dma->actual_len to zero
after urb actual length is updated.

Signed-off-by: Ravi Babu <ravibabu@ti.com>
---
 drivers/usb/musb/musb_host.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/musb/musb_host.c b/drivers/usb/musb/musb_host.c
index ead4796..d9659e6 100644
--- a/drivers/usb/musb/musb_host.c
+++ b/drivers/usb/musb/musb_host.c
@@ -1936,6 +1936,12 @@ finish:
 	urb->actual_length += xfer_len;
 	qh->offset += xfer_len;
 	if (done) {
+		/* Reset this dma ->actual_len after transfer complete,
+		 * if not subsequence dequeue request will take this
+		 */
+		if (dma)
+			dma->actual_len = 0;
+
 		if (use_sg)
 			use_sg = false;
 
-- 
1.7.0.4

